src/Dfe.PlanTech.AzureFunctions/Functions/QueueReceiver.cs

```diff
@@ -87,6 +92,45 @@ private async Task ProcessMessage(ServiceBusReceivedMessage message, ServiceBusM
         }
     }
 
+    private bool ShouldDropMessage(CmsEvent cmsEvent)
+    {
+        if (cmsEvent == CmsEvent.CREATE)
+        {
+            Logger.LogInformation("Dropping received event {cmsEvent}", cmsEvent);
+            return true;
+        }
+
+        if (ShouldIgnoreMessage(cmsEvent))
+        {
+            Logger.LogInformation("Receieved {event} but UsePreview is {usePreview} - dropping message", cmsEvent, _contentfulOptions.UsePreview);
+            return true;
+        }
+
+        return false;
+    }
+
+    private bool IsNewAndValidComponent(ContentComponentDbEntity mapped, ContentComponentDbEntity? existing)
```

Comment:
I think it would be good to call this method when updating an entity anyway. What if we receive an invalid update? 

This should be possible if, for example, a content designer edits an existing valid property, and makes it empty. This could autosave (thus triggering the webhook), meaning the update would fail.

It's not exactly an issue, as our data in the DB would still be valid, and the moment they fill in the field again all should be good. But if we're going through the effort of validating incoming data, I think we should do it fully.

Reply:
Resolved in: [180669b](https://github.com/DFE-Digital/plan-technology-for-your-school/pull/508/commits/180669b148cebadc97803a890dacca2408b8b97a)